src/instructlab/utils.py

```diff
@@ -530,27 +532,96 @@ def http_client(params):
-                    )
-                print("============================================================")
-        except Exception:
-            pass
-        return func(*args, **kwargs)
-
-    return wrapper
+    Returns name, value, type name, and parameter source
+    """
+    for key, value in kwargs.items():
+        src = ctx.get_parameter_source(key)
+        param_src: str = src.name.lower() if src is not None else "unknown"
+        type_name: str
+        if value is None:
+            type_name = "None"
+        else:
+            type_: type = type(value)
+            type_name = type_.__name__
+            mod_name: str | None = getattr(type_, "__module__", None)
+            if mod_name and mod_name != "builtins":
+                type_name = f"{mod_name}.{type_name}"
+        yield key, value, type_name, param_src
+    # additional args
+    if ctx.args:
+        yield "args", ctx.args, "list", "args"
+
+
+def display_params(f: typing.Callable) -> typing.Callable:
+    """Display command parameters decorator
+
+    The function decorator adds two hidden parameters to a click Command.
+
+    - `--display-params` dumps parameter info for humans and exits immediately.
+    - `--display-param-json` dumps parameter info as JSON and exits immediately.
+
+    When DEBUG logging is enabled, it also dumps parameter information for
+    humans, but does not exit immediately. The output contains parameter
+    name, value, value type (as qualified dotted string), and the source of
+    the value (commandline, environment, default, default_map, args,
+    unknown).
+    """
+
+    @wraps(f)
+    def wrapper(*args: typing.Any, **kwargs: typing.Any) -> typing.Any:
+        # some callbacks don't need ctx
+        ctx: click.Context | None = kwargs.get("ctx")
+        if ctx is None:
+            ctx = click.get_current_context()
+        display: str = kwargs.pop("_display_params")
+
+        # also display params when debug logging is enabled
+        if (
+            display is None
+            and logger.isEnabledFor(logging.DEBUG)
+            and not kwargs.get("quiet", False)
+        ):
+            display = "debug"
+
+        if display in {"debug", "human"}:
+            pad = min(max(len(name) for name in kwargs), 24)
+            print("Parameters:")
+            for name, value, typ, src in _get_param_info(ctx, **kwargs):
+                print(f"  {name:>{pad}}: {value!r:<8} \t[type: {typ} src: {src}]")
+        elif display == "json":
+            params = {
+                name: (value, type_name, param_src)
+                for name, value, type_name, param_src in _get_param_info(ctx, **kwargs)
+            }
+            json.dump({"params": params}, sys.stdout)
+
+        # --display-params* exit
+        if display in {"json", "human"}:
+            ctx.exit()
+
+        return f(*args, **kwargs)
+
+    human_option = click.option(
+        "--display-params",
```

Comment:
Given they are hidden flag intented for debugging I'm wondering it we shouldn't prefix them all with `--debug-` to make it more obvious. WDYT?